# Credit Card Customer Segmentation

## Goal

We are a data scients working in a bank. The goal is to segment the customers so that we can apply different business strategies to each. For example we could provide higher limits for customers that use the card a lot but spend little money. Or give incentives to those with high income that doesn't use it as much.<br>
We will apply the K-means algorithm to segment the data.<br>
The output should be a group for each client and an explanation of its characteristics.

## Libraries

In [4]:
import pandas as pd
import numpy as np
import seaborn as sns
from sklearn.cluster import KMeans  
from sklearn.preprocessing import StandardScaler

## Data Dictionary

- **customer_id**: unique identifier for each customer.
- **age**: customer age in years.
- **gender**: customer gender (M or F).
- **dependent_count**: number of dependents of each customer.
- **education_level**: level of education ("High School", "Graduate", etc.).
- **marital_status**: marital status ("Single", "Married", etc.).
- **estimated_income**: the estimated income for the customer projected by the data science team.
- **months_on_book**: time as a customer in months.
- **total_relationship_count**: number of times the customer contacted the company.
- **months_inactive_12_mon**: number of months the customer did not use the credit card in the last 12 months.
- **credit_limit**: customer's credit limit.
- **total_trans_amount**: the overall amount of money spent on the card by the customer.
- **total_trans_count**: the overall number of times the customer used the card.
- **avg_utilization_ratio**: daily average utilization ratio.

## Load data

In [6]:
ccc_df = pd.read_csv('data/credit_card_customers.csv')

## EDA

In [7]:
ccc_df.shape

(10127, 14)

In [8]:
ccc_df.head(5)

customer_id  age gender  dependent_count education_level marital_status  \
0    768805383   45      M                3     High School        Married   
1    818770008   49      F                5        Graduate         Single   
2    713982108   51      M                3        Graduate        Married   
3    769911858   40      F                4     High School        Unknown   
4    709106358   40      M                3      Uneducated        Married   

   estimated_income  months_on_book  total_relationship_count  \
0             69000              39                         5   
1             24000              44                         6   
2             93000              36                         4   
3             37000              34                         3   
4             65000              21                         5   

   months_inactive_12_mon  credit_limit  total_trans_amount  \
0                       1       12691.0                1144   
1                       1        8256.0                1291   
2                       1        3418.0                1887   
3                       4        3313.0                1171   
4                       1        4716.0                 816   

   total_trans_count  avg_utilization_ratio  
0                 42                  0.061  
1                 33                  0.105  
2                 20                  0.000  
3                 20                  0.760  
4                 28                  0.000

In [21]:
ccc_df['education_level'].value_counts(normalize=True)

education_level
Graduate         0.363879
High School      0.232152
Uneducated       0.173299
College          0.117705
Post-Graduate    0.060827
Doctorate        0.052138
Name: proportion, dtype: float64

### Observations

- There are 10k observations/customers
- We have 14 features
- Age, normally distributed min 26 max 73, avg 43
- Gender 53% female
- Avg dependant 2
- Most common education level is Graduate
- 46% are married, 39% single
- Income mean 62k, min 20k, max 200k
- Avg months on book is around 36 months
- Max inactivity in the last 12 mo is 6mo, the most common is 3mo
- Credit limit range [1438.3, 34,516], has a long tail distribution
- Total transactions amount [510, 18484], it has a long tail distribution